### PR TITLE
release: add bulk backport to build/release readme

### PR DESCRIPTION
**Pull request by BlaineEXE**
Add a bulk backport section to the readme in build/release.

Signed-off-by: Blaine Gardner <blaine.gardner@suse.com>

[skip ci]

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

File: build/release/README.md [hunk 1]
```diff
@@ -55,6 +55,47 @@ The tags allow for a progression of pre-releases such as:
...
+- **A :** `git log --format=oneline --no-merges --topo-order $RELEASE_BRANCH`
+- **B :** `git log --format=oneline --no-merges --topo-order master`
+
+Open a new window with the following view to list the commits which are part of the `master` branch
+and not part of the release branch. These will be in alphabetical sorted order rather than commit
+topological order.
+- **C :** `comm -1 -3 <(git log --format=format:'%s' --no-merges --topo-order release-1.2 | sort) <(git log --format=format:'%s' --no-merges --topo-order master | sort)`
+
+Open a fourth window to be our working window.
+1. `git co master`
```
Comment (travisn) [inline build/release/README.md]: is `co` your alias for `checkout`?

File: build/release/README.md [hunk 2]
```diff
@@ -55,6 +55,47 @@ The tags allow for a progression of pre-releases such as:
...
+2. `git co -b working-branch-for-$RELEASE_BRANCH` to make a working branch equivalent with `master`
+3. `git rebase --interactive $RELEASE_BRANCH` to rebase `master` changes on *top* of the release branch
+4. Compare the output in **C** to the commits here. git is not perfect and sometimes tries to rebase
```
Comment (travisn) [inline build/release/README.md]: If we need to check the commit list it's going to be error prone so I'm worried about this step. Did you find why git was trying to rebase those two commits for the backport earlier this week? Or you ended up just removing them from the commit history? It seems like something isn't right with our process if git is picking up extra commits.

File: build/release/README.md [hunk 3]
```diff
@@ -55,6 +55,47 @@ The tags allow for a progression of pre-releases such as:
...
+
+Open a new window with the following view to list the commits which are part of the `master` branch
+and not part of the release branch. These will be in alphabetical sorted order rather than commit
+topological order.
+- **C :** `comm -1 -3 <(git log --format=format:'%s' --no-merges --topo-order release-1.2 | sort) <(git log --format=format:'%s' --no-merges --topo-order master | sort)`
+
+Open a fourth window to be our working window.
+1. `git co master`
+2. `git co -b working-branch-for-$RELEASE_BRANCH` to make a working branch equivalent with `master`
+3. `git rebase --interactive $RELEASE_BRANCH` to rebase `master` changes on *top* of the release branch
```
Comment (travisn) [inline build/release/README.md]: When I tried the `git rebase --interactive` command last week it seemed that something wasn't quite right and the commit for the beta.0 tag in the manifests gave me a merge conflict. Then I fell back to do the bulk backport with these steps:
1. `git fetch --all`
1. `git checkout -b bulk-backport upstream/master`
1. `git push origin bulk-backport`
1. Open a PR to merge the branch to `release-1.2` branch
This didn't give me any merge conflicts and seemed like a simple way to get master back in sync with the release branch. Is there some advantage to the interactive rebase that I'm missing?

File: build/release/README.md (hunk 1 above)
Comment (BlaineEXE) [reply build/release/README.md]: Oh yeah. I should fix that :]

File: build/release/README.md (hunk 3 above)
Comment (BlaineEXE) [reply build/release/README.md]: I dunno. I don't think GitHub writes their own custom logic for PRs. I suspect that there would be the same double-commit issue I saw if I were to have done what you suggest, but I'm not sure. I can't speculate about any of this.

File: build/release/README.md (hunk 2 above)
Comment (BlaineEXE) [reply build/release/README.md]: `git` is a magic black box to me. I don't know why? Did anyone backport commits in between bulk backports?

File: build/release/README.md (hunk 3 above)
Comment (BlaineEXE) [reply build/release/README.md]: This is what happens if I attempt this strategy. I see a number of commits it's trying to add including the 2 rgw commits:
image

Comment (travisn) [reply build/release/README.md]: hmm I wonder what's up with those two commits? I can't anything special about them

**Comment by BlaineEXE** [comment]
I'm not doing any more work on this until the next release. This is the process I followed, which I document here, but I don't think I have time to budget for trying different techniques or figuring out what went wrong with the 2 commits.

**Pull request closed without merging**